spec/index.bs

```
@@ -1123,6 +1284,31 @@ and a |responseBody|, run the following steps. This returns an [=ordered map=].
     1. Return |json|.
 </div>
 
+<div algorithm>
+To <dfn>show an IDP login dialog</dfn> given an {{IdentityProviderAPIConfig}} |config|, run
+the following steps. This returns success or failure.
+    1. [=Create a fresh top-level traversable=] with URL
+        |config|.{{IdentityProviderAPIConfig/login_url}}.
+    1. The user agent MAY [=set up browsing context features=] or otherwise
+        affect the presentation of this traversable in an implementation-defined
+        way.
+    1. Wait for one of the following conditions:
+        * The user closes the browsing context: return failure.
+        * {{IdentityProvider}}.{{IdentityProvider/close}} is called in the
+            context of this new traversable:
+            1. Close the traversable.
+            1. Let |loginStatus| be the value of the entry in [=Login Status map=]
+                with the key being the origin of the {{IdentityProviderAPIConfig/login_url}}.
+
+                Note: The IDP login flow may set this value to logged-in using
+                    either the [[#login-status-javascript|JavaScript]] or
+                    [[#login-status-http|HTTP header]] API during the login
+                    flow. It is also possible that this change happened in
+                    a different browsing context.
+            1. If |loginStatus| is [=logged-in=], return success.
+            1. Otherwise, return failure.
```

Just wondering: shouldn't we restart the process instead of returning a failure?

---

Just curious:  does our implementation do? Do we retry?